osquery/tables/system/linux/cpu_time.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/tables.h>
+#include <osquery/filesystem.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kProcStat = "/proc/stat";
+
+std::vector<std::string> procFromFile(const std::string &path) {
+    std::string content;
+    std::vector<std::string> proc_lines;
+
+    if (!isReadable(path).ok()) {
+        return proc_lines;
+    }
+
+    if (!readFile(path, content).ok()) {
+        return proc_lines;
+    }
+
+    auto lines = split(content, "\n");
+
+    for (auto &line : lines) {
```

Comment:
Allocate your vector container right above the loop: `std::vector<std::string> proc_lines;`